client/dashboard/domains/domain-contact-details/region-address-fieldsets.tsx

```diff
@@ -0,0 +1,194 @@
+import { type StatesListItem, type DomainContactDetails } from '@automattic/api-core';
+import { __experimentalInputControl as InputControl, SelectControl } from '@wordpress/components';
+import { useViewportMatch } from '@wordpress/compose';
+import { type Field, type DataFormControlProps } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from 'react';
+import {
+	CHECKOUT_EU_ADDRESS_FORMAT_COUNTRY_CODES,
+	CHECKOUT_UK_ADDRESS_FORMAT_COUNTRY_CODES,
+	type CountryListItem,
+} from './custom-form-fieldsets/constants';
+import { getCountryPostalCodeSupport } from './custom-form-fieldsets/get-country-postal-code-support';
+
+const getPostalCodeLabel = ( countryCode: string ): string => {
+	switch ( countryCode ) {
+		case 'US':
+			return __( 'ZIP code' );
+		default:
+			return __( 'Postal Code' );
+	}
+};
+
+const getStateSelectLabel = ( countryCode: string ): string => {
```

Comment:
I don't see the select label in the UI. As a side effect, the first state is always selected. Is this expected? 

New UI:
<img width="639" height="130" alt="image" src="https://github.com/user-attachments/assets/17178d9a-06f4-4fa5-8c89-d386a2f0e835" />

Old UI:
<img width="522" height="100" alt="image" src="https://github.com/user-attachments/assets/899d3ac4-bb78-4911-a64d-004c7fba44f0" />

Reply:
That's a good question! Yes, it's indeed intentional. I see the same pattern in other areas of HD like `/dns/add` and `/forwarding/add`. Let's double-check it: p1759360568923139-slack-C0982082MSR